client/src/templates/Challenges/utils/frame.ts

```diff
@@ -267,9 +271,44 @@ const writeContentToFrame = (frameContext: Context) => {
     createHeader(frameContext.element.id) + frameContext.build,
     frameContext.document
   );
+
+  registerScrollEventListner(frameContext.element);
+
+  if (getPreviewScrollPosition()) {
+    restorePreviewScrollPosition(
+      frameContext.element,
+      getPreviewScrollPosition()
+    );
+  }
   return frameContext;
 };
 
+let previewScrollPosition = 0;
+export const getPreviewScrollPosition = () => previewScrollPosition;
+export const setPreviewScrollPosition = (position: number) =>
+  (previewScrollPosition = position);
+
+const registerScrollEventListner = (iframe: HTMLIFrameElement) => {
+  iframe.contentDocument?.addEventListener(
+    'scroll',
+    (event: IframeEvent<Document>) => {
```

Comment:
It looks like I can't even use the custom type in the listener function. Now, I can only think of using type guard.
CC @ShaunSHamilton

Reply:
I will push up the necessary changes.